src/processor/operator/persistent/node_batch_insert.cpp

```diff
@@ -198,10 +207,18 @@ void NodeBatchInsert::finalizeInternal(ExecutionContext* context) {
         }
     }
     if (nodeSharedState->globalIndexBuilder) {
-        nodeSharedState->globalIndexBuilder->finalize(context);
+        auto* localNodeState =
+            ku_dynamic_cast<BatchInsertLocalState*, NodeBatchInsertLocalState*>(localState.get());
+        KU_ASSERT(localNodeState->errorHandler.has_value());
+        nodeSharedState->globalIndexBuilder->finalize(context,
+            localNodeState->errorHandler.value());
+        localNodeState->errorHandler->flushStoredErrors();
```

Comment:
Just to double-check accessing the local node state during `finalize` shouldn't be a problem right? Otherwise I'll just create a new error handler instance.

Reply:
Should be fine here.